### PR TITLE
refactor divergence and created 'apply' subcommand

### PR DESCRIPTION
**Pull request by mogaal**
- Removed variables not in use (makefile)
- Update terraform version and deleted the auth0 provider (we are using the official one now)
- refactor(divergence): Consolidate structs in the terraform package and simplified code 
- feature(apply): Now you can do terraform apply with the CLI, maybe be   useful for future pipelines work

### CODE REVIEW

File: pkg/commands/terraform.go [hunk 1]
```diff
@@ -34,54 +25,49 @@ func addTerraformCmd(topLevel *cobra.Command) {
...
-		TfVarFile = append([]string{fmt.Sprintf("-var-file=%s", o.varFile)})
+			if err != nil {
+				contextLogger.Fatal("Error executing plan, either an error or a divergence")
+			}
+		},
 	}
 
-	contextLogger.Info("Executing terraform plan, if there is a drift this program execution will fail")
+	apply := &cobra.Command{
+		Use:    "apply",
```
Comment (jasonBirchall) [inline pkg/commands/terraform.go:40]: What's your thinking behind having Terraform apply here? I'm using wondering where it would be used..

**Review by jasonBirchall — CHANGES_REQUESTED** [review]
Overall it looks good. I have a few questions (below) and requests for change.

File: makefile [hunk 2]
```diff
@@ -1,6 +1,4 @@
 SOURCE_FILES := $(shell find * -name '*.go')
-DIRS_WITH_TESTS := $(shell find * -type f -name '*_test.go' | xargs -n 1 dirname | sort | uniq)
```
[the remark this answers is not in the document]
Comment (mogaal) [reply makefile:2]: `make test` will still working in the same way. It isn't actually using these variables but it is doing what you mention (`go test ...`). Have a look at [makefile in main branch](https://github.com/ministryofjustice/cloud-platform-cli/blob/main/makefile#L10-L11)

File: makefile [hunk 3]
```diff
@@ -1,6 +1,4 @@
...
-DIRS_WITH_GOFILES := $(shell find * -type f -name '*.go' | xargs -n 1 dirname | sort | uniq)
```
[the remark this answers is not in the document]
Comment (mogaal) [reply makefile:3]: Same above

File: pkg/commands/terraform.go (hunk 1 above)
Comment (mogaal) [reply pkg/commands/terraform.go:40]: The idea I have on my mind is to add a second step after the [divergence pipeline](https://concourse.cloud-platform.service.justice.gov.uk/teams/main/pipelines/divergence) which can be triggered manually and would run `terraform apply`. Similar to what they document [here](https://concourse-ci.org/manual-trigger-example.html)

File: .github/workflows/docker-hub.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+      - name: Push tools image to docker hub
+        uses: docker/build-push-action@v1
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+          repository: ministryofjustice/cloud-platform
```
[the remark this answers is not in the document]
Comment (mogaal) [reply .github/workflows/docker-hub.yml:19]: Ohh I didn't see this action. You are right, I am deleting this file now.

File: pkg/commands/terraform.go (hunk 1 above)
Comment (mogaal) [reply pkg/commands/terraform.go:40]: I also think in the near future, when we start to execute everything in a pipeline, we will need to execute terraform apply in different places. We already have the CLI which does a plan (divergence is just a plan).

File: makefile (hunk 2 above)
[the remark this answers is not in the document]
Comment (jasonBirchall) [reply makefile:2]: Agreed. I think I misread the change being made.

File: pkg/commands/terraform.go (hunk 1 above)
Comment (jasonBirchall) [reply pkg/commands/terraform.go:40]: Sounds good to me.

**Review by jasonBirchall — APPROVED** [review]
:+1:

**Pull request merged**